server/src/main/java/org/elasticsearch/cluster/metadata/MetadataIndexAliasesService.java

```diff
@@ -244,4 +241,73 @@ private static void validateAliasTargetIsNotDSBackingIndex(ClusterState currentS
             );
         }
     }
+
+    /**
+     * A cluster state update task that applies the alias actions to the given cluster state.
+     */
+    record ApplyAliasActions(
+        MetadataIndexAliasesService aliasesService,
+        IndicesAliasesClusterStateUpdateRequest request,
+        ActionListener<AcknowledgedResponse> listener
+    ) implements ClusterStateTaskListener {
+
+        @Override
+        public void onFailure(Exception e) {
+            listener.onFailure(e);
+        }
+
+        ClusterStateAckListener getAckListener() {
```

Comment:
Nit: could we have `ApplyAliasActions` implement `ClusterStateAckListener` and then just use ourselves instead of having to create a separate listener?

Reply:
Let me give it a try.

Reply:
Much nicer :)